### ISSUE TITLE
Allow adding individual studies/series to the cohort

### ISSUE THREAD

**Issue by fedorov** [ImagingDataCommons/IDC-WebApp#419]
As communicated by Ulli, "Are there plans to allow users to add selected studies or series to the cohort (not just collections or cohorts created through filtering)". This is the capability that is available in TCIA.

**Comment by fedorov** [comment]
Agreed that it makes sense to implement selection modes similar to how TCIA does it with colors coding complete or partial inclusion. The selected UIDs will be adjusting the definition of the filter. This will be a significant UI effort, but should be easy on the back-end, per @s-paquette.

**Comment by G-White-ISB** [comment]
I believe thinking is that included/excluded lists ARE part of the cohort. To be discussed

**Comment by fedorov** [comment]
Based on the discussion today, we agreed this is most likely a post-production feature.

We need to reach consensus on how this feature should be implemented, document the plan in this issue, and then present the plan to the users (specifically, Brad Erickson lab) to get their opinion before this is implemented.

**Comment by G-White-ISB** [comment]
If a user is selecting a UID for inclusion or exclusion that implies they have some information about the pertaining study or series that is not exposed in the web app.  Now it's true this could be subjective ('i.e. I think this image is fuzzy') but in other cases we may need wish to guide the Web App based on the user's criteria, (i.e. I wish to exclude these UIDs because I know these patients were male).

**Comment by fedorov** [comment]
Per discussion today, the plan to implement this feature is to keep the current organization of the interface with collection/patient/study/series tables, and add some mechanism for each of the rows in those respective tables to include them into a "shopping cart" equivalent (we need to agree on the name). 

There will be a limit in the order of a 900 on the number of items that one can select. The limit is applied at the level of the individual hierarchy level (i.e., one will be able to select up to 900 patients AND up to 900 series etc).

**Comment by G-White-ISB** [comment]
Per recent discussions, the plan is to maintain the selections in the cart even after the user has updated the filter. This is going to introduce some technical challenges. Currently the web does not maintain a cart or manifest list. It can be constructed from the filter itself. If the user selections are going to persist after a filter update, then I believe the web app will be forced to maintain a list of the cases in the cart , and to retrieve all matching cases when the filter is changed. 

Example: suppose a user filters for **COLLECTION in (LIDC-IDRI) and MODALITY IN (Segmentation)** and adds the LIDC-IDRI collection to the cart. As currently implemented the 875 cases in LIDC-IDRI with segmentation will be added to the cart. If the filter then changes to just **COLLECTION in (LIDC-IDRI)** then the webapp will have to track which cases in LIDC-IDRI are already in the cart and which are not. This is only an issue with selections at the collection level, since cases, studies, and series are never partially selected by a filter.

**Comment by fedorov** [comment]
Based on the review of the prototype:
* I think the implementation of this feature should not add any additional vertical space to the portal interface - the details about filtersets should be hidden behind some button, since the interface is already extremely complex and it takes too long to get to the data, and also I am absolutely convinced most users will not care about filtersets
* I would like to see the entire implementation that includes manifest generation and download, since the key question is how much complexity and dealays those filtersets will add and how it will affect user experience
* I hope that appearance of the buttons will be refined in the implementation for the user release

I also maintain my concern that the precedent of the existing CRDC nodes is that users get simple, intuitive cart. No filtersets:
* CDS portal: https://dataservice.datacommons.cancer.gov/#/data
* GDC does have a simple cart: https://portal.gdc.cancer.gov/repository
* ICDC has "file-centric cart": https://caninecommons.cancer.gov/#/fileCentricCart

As long as the filterset business is transparent to the user, doesn't increase complexity and does not degrade performance, it is at this point, but otherwise it will be a problem.

Whenever we have the complete prototype (the one that queries and gives user the manifest), we should evaluate and discuss with the group.

**Comment by G-White-ISB** [comment]
@fedorov I take it you are not a fan of auto-creating a filterset upon change in the filter, or a dialog opening to ASK if you want to create a new filter set upon change in the filter. I concede the cart selection in the other portals are consistent with each other and simpler than what I had shown you. Our thinking was to maintain the 'provenance' of the selection, to show the user what filter criteria they were using when they added something to the cart.

 If we don't auto-create filtersets in response to cart selections then filtersets are a separate issue.

**Comment by wlongabaugh** [comment]
@fedorov I think we should be able to make filterset creation completely hidden, with only a single dropdown or button in the current filter display panel that would be the only evidence that it exists.

**Comment by fedorov** [comment]
Also, to avoid miscommunication, I am strongly against showing this popup to the user.

![image](https://github.com/ImagingDataCommons/IDC-WebApp/assets/313942/64083106-d5d7-473c-bc21-9c8befc9946e)

Please pick the default behavior in such a way that the functionality delivered to the user is equivalent to the behavior of the cart in all other systems known to the mankind. If you absolutely want to have this setting somewhere, please hide it behind a button that does not add any vertical space to the page.
